Sources/WalletConnectAuth/Types/Session/WCSession.swift

```diff
@@ -62,47 +62,46 @@ struct WCSession: ExpirableSequence {
         return controller.publicKey == peerParticipant.publicKey
     }
     
-    // FIXME
     func hasNamespace(for chain: Blockchain) -> Bool {
-        // TODO
-//        namespaces.contains{$0.chains.contains(chain)}
-        return true
+        return namespaces[chain.namespace] != nil
     }
     
-    // TODO: Remove optional for chain param, it's required now / protocol change
-    func hasNamespace(for chain: Blockchain?, method: String) -> Bool {
-        // TODO
-//        if let chain = chain {
-//            let namespacesIncludingChain = namespaces.filter{$0.chains.contains(chain)}
-//            let methods = namespacesIncludingChain.flatMap{$0.methods}
-//            return methods.contains(method)
-//        } else {
-//            return namespaces
-//                .filter { $0.chains.isEmpty }
-//                .flatMap { $0.methods }
-//                .contains(method)
-//        }
-        return true
+    func hasPermission(forMethod method: String, onChain chain: Blockchain) -> Bool {
+        if let namespace = namespaces[chain.namespace] {
+            if namespace.accounts.contains(where: { $0.blockchain == chain }) {
+                if namespace.methods.contains(method) {
+                    return true
+                }
+            } else if let extensions = namespace.extensions {
+                for extended in extensions {
+                    if extended.accounts.contains(where: { $0.blockchain == chain }) {
+                        if extended.methods.contains(method) {
+                            return true
+                        }
```

Comment:
I am not sure it is correct, are you?
I think that if namespace.acconts has not a specific chain, extension will not have as well 🤔 

lets unit test this

Reply:
Implementations fixed to follow the namespace specs